tests/testthat/test-summary.pop_data.R

```diff
@@ -0,0 +1,92 @@
+xs_data <- load_pop_data(
+  file_path = "https://osf.io/download//n6cp3/",
+  age = "Age",
+  value = "result",
+  id = "index_id",
+  standardize = TRUE
+)
+
+unstratified_summary <- xs_data %>%
+  summary() %>%
+  magrittr::extract2("age_summary")
+
+test_that("`summary.pop_data()` produces same results when stratified", {
+  min_country <- xs_data %>%
+    summary(strata = "Country") %>%
+    magrittr::extract2("age_summary") %>%
+    pull(age_min) %>%
+    min()
+
+  expect_equal(object = unstratified_summary %>% pull(age_min), expected = min_country)
+})
+
+test_that("`summary.pop_data()` produces same results when stratified", {
+  first_quartile_country <- xs_data %>%
+    summary(strata = "Country") %>%
+    magrittr::extract2("age_summary") %>%
+    pull(age_first_quartile) %>%
+    mean()
+
+  expect_equal(
+    unstratified_summary %>% pull(age_first_quartile) %>% as.numeric(),
+    first_quartile_country,
+    tolerance = 0.5
+  )
```

Comment:
as discussed in teams, we want to test:

does the function execute without errors for standard input values (`strata = NULL`, strata = "Country", etc)
does the function return what we expect it to for various input values? (pre-save some example outputs as internal data, and check for equality of current outputs)

we don't expect stratified and unstratified results to be similar to each other.

Reply:
I have written new tests to replace these.